### ISSUE TITLE
Make SpatialRule more flexible

### ISSUE THREAD

**Issue by boldtrn** [graphhopper/graphhopper#1427]
After working with SpatialRules for some time I have seen a couple of issues with SpatialRules being not flexible enough.

For example SpatialRules have the method:

```
public double getMaxSpeed(String highwayTag, double _default)
``` 

There are cases where only the highwayTag is not enough to determine the maxSpeed of a country. One current example would be the new maxspeed rules in France, [where the maxspeed depends on the number of lanes](https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Maxspeed#France).

Therefore I propose to change the SpatialRule method signatures to:

```
public double getMaxSpeed(Readerway way, double _default)
``` 

This makes the handling a lot more flexible. The same applies to the method `getAccessValue`, which could include local differences. See for example #1408 for a good example.

So I would also like to change the method signature of `getAccessValue` to use the way instead of only the highway.

WDYT?

**Comment by easylite376** [comment]
+1 from me.

**Comment by easbar** [comment]
This was fixed in #2353 or even before.
